packages/cli/src/sso/saml/routes/saml.controller.protected.ee.ts

```diff
@@ -96,12 +120,23 @@ samlControllerProtected.get(
 	SamlUrls.initSSO,
 	samlLicensedAndEnabledMiddleware,
 	async (req: express.Request, res: express.Response) => {
-		const url = SamlService.getInstance().getRedirectLoginRequestUrl();
-		if (url) {
-			// TODO:SAML: redirect to the URL on the client side
-			return res.status(301).send(url);
+		const result = SamlService.getInstance().getLoginRequestUrl();
+		if (result?.binding === 'redirect') {
+			// forced client side redirect
+			return res.send(getInitSSOPostView(result.context));
+			// return res.status(301).send(result.context.context);
+		} else if (result?.binding === 'post') {
+			return res.send(getInitSSOFormView(result.context as PostBindingContext));
 		} else {
 			throw new AuthError('SAML redirect failed, please check your SAML configuration.');
 		}
 	},
 );
+
+samlControllerProtected.get(
+	SamlUrls.configTest,
+	async (req: express.Request, res: express.Response) => {
+		const testResult = await SamlService.getInstance().testSamlConnection();
```

Comment:
Yes this is just a very basic implementation for the moment, until I know how the frontend wants to call it. Right now it's a return of true/false for the test, rather than an exception. The entire test process may change, since it may be better to have the owner go through a full sso flow instead to confirm it is working.